integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,288 @@
+            A custom query for the embedding retriever.
+
+        :param join_mode:
+            The mode to use for joining the results from the BM25 and embedding retrievers.
+
+        :param weights:
+            The weights for the joiner.
+
+        :param top_k:
+            The number of results to return from the joiner.
+
+        :param sort_by_score:
+            Whether to sort the results by score.
+
+        :param **kwargs:
+            Additional keyword arguments.
+        """
+        self.document_store = document_store
+        self.embedder = embedder
+
+        # OpenSearchBM25Retriever
+        self.filters_bm25 = filters_bm25
+        self.fuzziness = fuzziness
+        self.top_k_bm25 = top_k_bm25
+        self.scale_score = scale_score
+        self.all_terms_must_match = all_terms_must_match
+        self.filter_policy_bm25 = filter_policy_bm25
+        self.custom_query_bm25 = custom_query_bm25
+
+        # OpenSearchEmbeddingRetriever
+        self.filters_embedding = filters_embedding
+        self.top_k_embedding = top_k_embedding
+        self.filter_policy_embedding = filter_policy_embedding
+        self.custom_query_embedding = custom_query_embedding
+
+        # DocumentJoiner
+        self.join_mode = join_mode
+        self.weights = weights
+        self.top_k = top_k
+        self.sort_by_score = sort_by_score
+
+        init_args = {
+            "bm25_retriever": {
+                "document_store": self.document_store,
+                "filters": self.filters_bm25,
+                "fuzziness": self.fuzziness,
+                "top_k": self.top_k_bm25,
+                "scale_score": self.scale_score,
+                "all_terms_must_match": self.all_terms_must_match,
+                "filter_policy": self.filter_policy_bm25,
+                "custom_query": self.custom_query_bm25,
+            },
+            "embedding_retriever": {
+                "document_store": self.document_store,
+                "filters": self.filters_embedding,
+                "top_k": self.top_k_embedding,
+                "filter_policy": self.filter_policy_embedding,
+                "custom_query": self.custom_query_embedding,
+            },
+            "document_joiner": {
+                "join_mode": self.join_mode,
+                "weights": self.weights,
+                "top_k": self.top_k,
+                "sort_by_score": self.sort_by_score,
+            },
+        }
+
+        self.extra_args = kwargs
+
+        # handle extra kwargs for the bm25 and embedding retrievers and the doc store as init param
+        if "bm25_retriever" in kwargs:
+            init_args["bm25_retriever"].update(kwargs["bm25_retriever"])
+            init_args["bm25_retriever"]["document_store"] = self.document_store
+        if "embedding_retriever" in kwargs:
+            init_args["embedding_retriever"].update(kwargs["embedding_retriever"])
+            init_args["embedding_retriever"]["document_store"] = self.document_store
```

Comment:
It might be nice to run a check on kwargs as in if we notice the user passed something like 
```python
kwargs = {"fuzzines": "AUTO"}
```
we could warn them saying `kwargs must contain 'bm25_retriever' and 'embedding_retriever' as top-level keys ...`